### PR TITLE
ARROW-16595: [WEBSITE] DataFusion 8.0.0 release blog post

### PR DESCRIPTION
**Pull request by andygrove**
Based on crowd-sourced draft at https://docs.google.com/document/d/16PhXQs4s2HJNPLydy5J5pqXnefI-iR3VxGqrO497M08/edit?usp=sharing

### CODE REVIEW

**Comment by andygrove** [comment]
@alamb @matthewmturner @yjshen fyi

**Comment by matthewmturner** [comment]
Maybe it's also worth mentioning how InfluxDB is going to publish their implementation of `ObjectStore` for the community to use - which will provide a well tested and actively used implementation and support for s3, Azure, and GCP object stores.

ill defer to @alamb on that.

Do you also think we could mention new projects using DataFusion - i.e. what you are doing with Dask? Or is it too soon to announce that?

**Comment by andygrove** [comment]
I'm not against mentioning dask-sql but it seemed separate to the announcement of the new features. I think it might even be worth a separate blog post on that once the work is merged to master in dask-sql.

**Comment by alamb** [comment]
Here are some proposed changes: https://github.com/andygrove/arrow-site/pull/1

**Comment by alamb** [comment]
Thanks for taking the initiative on this one @andygrove ❤️

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
> Maybe it's also worth mentioning how InfluxDB is going to publish their implementation of ObjectStore for the community to use - which will provide a well tested and actively used implementation and support for s3, Azure, and GCP object stores.


I tried to add something related to `ObjectStore` consolidation in https://github.com/andygrove/arrow-site/pull/1

I wonder if we should perhaps add links to the DataFusion issues to draw additional attention to it 🤔  -- if people think that is a good idea I can try and find time to make the links

**Comment by andygrove** [comment]
> Here are some proposed changes: [andygrove#1](https://github.com/andygrove/arrow-site/pull/1)

Thanks @alamb I have merged the PR

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+# Introduction
+
+[DataFusion](https://arrow.apache.org/datafusion/) is an extensible query execution framework, written in Rust, that
+uses Apache Arrow as its in-memory format.
+
+When you want to extend your Rust project with [SQL support](https://arrow.apache.org/datafusion/user-guide/sql/sql_status.html),
+a DataFrame API, or the ability to read and process Parquet, JSON, Avro or CSV data, DataFusion is definitely worth
+checking out.
+
+DataFusion's  SQL, `DataFrame`, and manual `PlanBuilder` API let users access a sophisticated query optimizer and
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:36]:
```suggestion
DataFusion's SQL, `DataFrame`, and manual `PlanBuilder` API let users access a sophisticated query optimizer and
```

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+execution engine capable of fast, resource efficient, and parallel execution that takes optimal advantage of
+today's multicore hardware. Being written in Rust means DataFusion can offer *both* the safety of dynamic languages and
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:38]:
```suggestion
today's multicore hardware. Being written in Rust means DataFusion can offer *both* the safety of a dynamic language and
```

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+changes also make DataFusion an increasingly compelling choice for
+projects looking for a SQL parser and query planner to translate to
+their own execution engine.
+
+## Query Execution & Internals
+
+There are several notable improvements and new features in the query execution engine:
+
+- The `ExecutionContext` has been renamed to `SessionContext` and now supports multi-tenancy
+- The `ExecutionPlan` trait is no longer `async`
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:145]: 👍

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+- A new serialization API for serializing plans to bytes (based on protobuf)
+
+In addition, we have added several foundational features to drive even
+more advanced query processing into DataFusion, focused on running
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:149]:
```suggestion
more advanced query processing into DataFusion, focusing on running
```

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+arbitrary queries larger than available memory, and pushing the
+envelope for performance of sorting, grouping and joining even
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:151]:
```suggestion
envelope for performance of sorting, grouping, and joining even
```

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+further:
+
+- Morsel-Driven Scheduler based on ["Morsel-Driven Parallelism: A NUMA-Aware Query
+  Evaluation Framework for the Many-Core Age"](https://15721.courses.cs.cmu.edu/spring2016/papers/p743-leis.pdf)
+- Consolidated object store implementation and integration with parquet decoding
+- Memory Limited Spilling Sort operator
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:157]:
```suggestion
- Memory Limited Spilling sort operator
```

File: _posts/2022-05-16-datafusion-8.0.0.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+- There is a [proposal to move Ballista to its own top-level arrow-ballista repository](https://docs.google.com/document/d/1jNRbadyStSrV5kifwn0khufAwq6OnzGczG4z8oTQJP4/edit?usp=sharing)
+ to decouple DataFusion and Ballista releases and to allow each project to have documentation better targeted at
+  its particular audience.
+- We plan on increasing the frequency of DataFusion releases, with monthly releases now instead of quarterly. This
+  is driven by requests from the increasing number of projects that now depend on DataFusion.
+- There is ongoing work to implement new optimizer rules to rewrite queries containing subquery expressions as
+  joins, to support a wider range of queries.
+- The new scheduler based on morsel-driven execution will continue to evolve in this next release, with work to
+ refine IO abstractions to improve performance and integration with the new scheduler.
```
Comment (Jimexist) [inline _posts/2022-05-16-datafusion-8.0.0.md:185]:
```suggestion
  refine IO abstractions to improve performance and integration with the new scheduler.
```

**Review by Jimexist — COMMENTED** [review]
i left a few minor comments but the contents are looking good 👍

**Review by yjshen — APPROVED** [review]

**Comment by alamb** [comment]
🎉

**Pull request merged**